sklearn/utils/mocking.py

```diff
@@ -55,7 +58,7 @@ def fit(self, X, y):
     def predict(self, T):
         if self.check_X is not None:
             assert_true(self.check_X(T))
-        return T.shape[0]
+        return np.zeros(_num_samples(T))
```

Comment:
To minimize future surprise, I think we should make it a real classifier and implement `classes_` as we do in other classifiers. Maybe also add the `ClassifierMixin`.